Functions/QualityControl/xASL_qc_CleanOldQC.m

```diff
@@ -0,0 +1,94 @@
+function [x] = xASL_qc_CleanOldQC(x, ScanType, bRemoveCurrentSession)
+%xASL_qc_RemoveOutdatedQC Removes outdated QC parameters, loads QC from JSON and cleans the current session if necessary
+%
+% FORMAT: [x] = xASL_qc_RemoveOutdatedQC(x, ScanType[, bRemoveCurrentSession)
+%
+% INPUT:
+%   x 	                  - structure containing fields with all information required to run this submodule (REQUIRED)
+%   ScanType              - Type of the data - can be 'ASL' or 'func' (REQUIRED)
+%   bRemoveCurrentSession - Removes the current session completely (OPTIONAL, DEFAULT = false)
+%
+% OUTPUT:
+%   x        - same as input
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION: This function removes the old and outdated QC parameters. Specifically parameters for ASL and func which are now given in a subfield by session.
+%              It also loads further sessions from the disc and if necessary, it cleans the current session completely.
+%
+% EXAMPLE: x = xASL_qc_RemoveOutdatedQC(x, 'ASL');
+% __________________________________
+% Copyright (C) 2015-2023 ExploreASL	
+
+if nargin < 2
+	error('Two parameters are required');
+end
+
+if nargin < 3 || isempty(bRemoveCurrentSession)
+	bRemoveCurrentSession = false;
+end
+
+% All possible scantype lists
+ScanTypeList = {'func', 'ASL', 'dwi'};
+
+% Load QC.json and save it to x-struct
+QC_Path = fullfile(x.D.ROOT, x.SUBJECT, ['QC_collection_' x.SUBJECT '.json']);
+if xASL_exist(QC_Path, 'file')
+	% Load the save QC parameters
+	oldOutput = xASL_io_ReadJson(QC_Path);
+
+	% Go through all possible scantypes
+	for iScanType = 1:length(ScanTypeList)
+		currentScanType = ScanTypeList{iScanType};
+		if isfield(oldOutput, currentScanType)
+			% Load all subfields for that scantype
+			listFields = fieldnames(oldOutput.(currentScanType));
+			for iField = 1:length(listFields)
+				% Copy only the subfields in ScanType_N
+				if ~isempty(regexp(listFields{iField}, [currentScanType '_\d+'], 'once'))
+					% But skip the current session unless bCompleteRun is false
+					if ~strcmp(listFields{iField}, x.SESSION) || ~bRemoveCurrentSession
+						x.Output.(currentScanType).(listFields{iField}) = oldOutput.(currentScanType).(listFields{iField});
```

Comment:
Can't we do here exactly the same for x.Output_im as x.Output, because those are designed to have similar structure?

Reply:
Looks great for output/QC.json. But don't we want to do the same for x.output_im? I forgot your reason to do this, but this would make sense. Currently, we use the following general function for this: `xASL_vis_AddIM2QC` within the structural and ASL modules. So, you would only have to change this function accordingly.

Reply:
Not sure if I follow.
We can't do the same for output as for output_im. As output is saved in QC.json and we can't save images to QC.json. I'm happy to change things and I'm not disagreeing, I just don't understand what are you asking me now to do...

Reply:
Both output and output_im are stored likewise in x.mat, with identical struct naming conventions. The difference is that x.output also goes to the QC.json. 

You changed the struct naming convention for x.output. I consider that we change the naming convention similarly for x.output_im.